library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java

```diff
@@ -654,6 +673,7 @@ public TimeBasedDirectoryScanner()
       discoveredFiles = new LinkedBlockingDeque<FileInfo>();
       atomicThrowable = new AtomicReference<Throwable>();
       ignoredFiles = Sets.newHashSet();
+      lock = new Lock();
```

Comment:
If we initialize it here and don't expect the object to change, can declare it final. Don't see why it should be moved to setup()?